projects/core/src/product/facade/product-availability.service.spec.ts

```diff
@@ -0,0 +1,76 @@
+import { TestBed } from '@angular/core/testing';
+import { Command, CommandService } from '../../../src/util/command-query';
+import { of } from 'rxjs';
+import { ProductAvailabilities } from '../../model/product.model';
+import { ProductAvailabilityConnector } from '../connectors';
+import { ProductAvailabilityService } from './product-availability.service';
+
+describe('ProductAvailabilityService', () => {
+  let service: ProductAvailabilityService;
+  let connector: jasmine.SpyObj<ProductAvailabilityConnector>;
+  let commandService: jasmine.SpyObj<CommandService>;
+  let commandSpy: jasmine.SpyObj<Command<any, ProductAvailabilities>>;
+
+  beforeEach(() => {
+    commandSpy = jasmine.createSpyObj('Command', ['execute']);
+    connector = jasmine.createSpyObj('ProductAvailabilityConnector', [
+      'getRealTimeStock',
+    ]);
+    commandService = jasmine.createSpyObj('CommandService', ['create']);
+
+    commandService.create.and.returnValue(commandSpy);
+
+    TestBed.configureTestingModule({
+      providers: [
+        ProductAvailabilityService,
+        { provide: ProductAvailabilityConnector, useValue: connector },
+        { provide: CommandService, useValue: commandService },
+      ],
+    });
+
+    service = TestBed.inject(ProductAvailabilityService);
+  });
+
+  it('should be created', () => {
+    expect(service).toBeTruthy();
+  });
+
+  describe('getRealTimeStock', () => {
+    it('should call getRealTimeStockCommand.execute with correct parameters', () => {
+      const productCode = 'testProductCode';
+      const sapCode = 'testSapCode';
+      const expectedStockData: ProductAvailabilities = {
+        quantity: '100',
+        status: 'IN_STOCK',
+      };
+
+      connector.getRealTimeStock.and.returnValue(of(expectedStockData));
+
+      commandSpy.execute.and.returnValue(of(expectedStockData));
+
+      service.getRealTimeStock(productCode, sapCode).subscribe((stockData) => {
+        expect(commandSpy.execute).toHaveBeenCalledWith({
+          productCode,
+          sapCode,
+        });
+        expect(stockData).toEqual(expectedStockData);
```

Comment:
[MINOR]
Let's call `done()` after this line - because it's inside the `subscription()`.
Otherwise, if the observable didn't emit, no expect() statement would run and test would pass - giving a false positive result.

Reply:
Done